streams/src/test/java/org/apache/kafka/streams/kstream/internals/KStreamImplTest.java

```diff
@@ -2330,4 +2348,266 @@ public void shouldNotAllowNullNamedOnProcessWithStores() {
         assertThat(exception.getMessage(), equalTo("named can't be null"));
     }
 
+
+    @Test
+    public void shouldKTableFromKStream() {
+        final Consumed<String, String> consumed = Consumed.with(Serdes.String(), Serdes.String());
+
+        final StreamsBuilder builder = new StreamsBuilder();
+
+        final String input = "input";
+        final String output = "output";
+
+        builder.stream(input, consumed).toTable().toStream().to(output);
+
+        try (final TopologyTestDriver driver = new TopologyTestDriver(builder.build(), props)) {
+            final TestInputTopic<String, String> inputTopic =
+                driver.createInputTopic(input, Serdes.String().serializer(), Serdes.String().serializer());
+            final TestOutputTopic<String, String> outputTopic =
+                driver.createOutputTopic(output, Serdes.String().deserializer(), Serdes.String().deserializer());
+
+            inputTopic.pipeInput("A", "01", 5L);
+            inputTopic.pipeInput("B", "02", 100L);
+            inputTopic.pipeInput("C", "03", 0L);
+            inputTopic.pipeInput("D", "04", 0L);
+            inputTopic.pipeInput("A", "05", 10L);
+            inputTopic.pipeInput("A", "06", 8L);
+
+            final List<TestRecord<String, String>> outputExpectRecords = new ArrayList<>();
+            outputExpectRecords.add(new TestRecord<>("A", "01", Instant.ofEpochMilli(5L)));
+            outputExpectRecords.add(new TestRecord<>("B", "02", Instant.ofEpochMilli(100L)));
+            outputExpectRecords.add(new TestRecord<>("C", "03", Instant.ofEpochMilli(0L)));
+            outputExpectRecords.add(new TestRecord<>("D", "04", Instant.ofEpochMilli(0L)));
+            outputExpectRecords.add(new TestRecord<>("A", "05", Instant.ofEpochMilli(10L)));
+            outputExpectRecords.add(new TestRecord<>("A", "06", Instant.ofEpochMilli(8L)));
+
+            assertEquals(outputTopic.readRecordsToList(), outputExpectRecords);
+        }
+    }
+
+
+    @Test
+    public void shouldKTableFromKStreamCheckMaterializedStore() {
```

Comment:
nit (rename): `shouldMaterializeKTableFromKStream`

(a good naming patter is "should <doSomething> <condition/operation>"